refinery/tool_manager/utils.py

```diff
@@ -128,27 +136,53 @@ def create_tool_definition(annotation_data):
         workflow_engine = WorkflowEngine.objects.get(
             uuid=annotation_data["workflow_engine_uuid"]
         )
+        workflow = WorkflowFactory(
+            uuid=str(uuid.uuid4()),
+            name=annotation_data["name"],
+            summary="Workflow for: {}".format(annotation_data["name"]),
+            internal_id=annotation_data["galaxy_workflow_id"],
+            workflow_engine=workflow_engine,
+            is_active=True,
+            type=Workflow.ANALYSIS_TYPE,
+            graph=json.dumps(annotation_data["graph"])
+        )
+        workflow.set_manager_group(workflow_engine.get_manager_group())
+        workflow.share(workflow_engine.get_manager_group().get_managed_group())
 
         tool_definition = ToolDefinitionFactory(
             name=annotation_data["name"],
             description=annotation["description"],
             tool_type=tool_type,
             file_relationship=create_file_relationship_nesting(annotation),
-            galaxy_workflow_id=annotation_data["galaxy_workflow_id"],
-            workflow_engine=workflow_engine
+            workflow=workflow
         )
     elif tool_type == ToolDefinition.VISUALIZATION:
         tool_definition = ToolDefinitionFactory(
             name=annotation_data["name"],
             description=annotation["description"],
             tool_type=tool_type,
-            file_relationship=create_file_relationship_nesting(
-                annotation
-            ),
+            file_relationship=create_file_relationship_nesting(annotation),
             image_name=annotation["image_name"],
             container_input_path=annotation["container_input_path"],
         )
 
+        # If we've successfully created the ToolDefinition lets
+        #  pull down its docker image
+        try:
+            image_name, version = tool_definition.image_name.split(":")
+        except ValueError:
+            logger.debug(
+                "`image_name` has no specified version, and the `latest` "
+                "version of %s will be pulled",
+                tool_definition.image_name
+            )
+            DockerClientWrapper().pull(tool_definition.image_name)
```

Comment:
Oh: I would not continue here: I don't think the default behavior is ever helpful, and debug level logging may very well not be turned on .